dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/trackedentity/store/mapper/EventRowCallbackHandler.java

```diff
@@ -0,0 +1,104 @@
+ */
+
+package org.hisp.dhis.dxf2.events.trackedentity.store.mapper;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.vividsolutions.jts.geom.Geometry;
+import org.hisp.dhis.category.CategoryOption;
+import org.hisp.dhis.dxf2.events.enrollment.EnrollmentStatus;
+import org.hisp.dhis.dxf2.events.event.Event;
+import org.hisp.dhis.event.EventStatus;
+import org.hisp.dhis.organisationunit.FeatureType;
+import org.hisp.dhis.util.DateUtils;
+import org.postgresql.util.PGobject;
+
+import static org.hisp.dhis.dxf2.events.trackedentity.store.mapper.MapperGeoUtils.resolveGeometry;
+
+/**
+ * @author Luciano Fiandesio
+ */
+public class EventRowCallbackHandler
+    extends
+    AbstractMapper<Event>
+{
+
+    @Override
+    Event getItem( ResultSet rs )
+        throws SQLException
+    {
+        return getEvent( rs );
+    }
+
+    @Override
+    String getKeyColumn() {
+        return "enruid";
+    }
+
+    private Event getEvent( ResultSet rs )
+        throws SQLException
+    {
+        Event event = new Event();
+        event.setEvent( rs.getString( "uid" ) );
+        event.setId( rs.getLong("programstageinstanceid") );
+        event.setTrackedEntityInstance( rs.getString( "enruid" ) );
+        event.setFollowup( rs.getBoolean( "enrfollowup" ) );
+        event.setEnrollmentStatus( EnrollmentStatus.fromStatusString( rs.getString("enrstatus") ) );
+        event.setStatus( EventStatus.valueOf( rs.getString( "status" ) ) );
+        event.setEventDate( DateUtils.getIso8601NoTz( rs.getDate( "executiondate" ) ) );
+        event.setDueDate( DateUtils.getIso8601NoTz( rs.getDate( "duedate" ) ) );
+        event.setStoredBy( rs.getString( "storedby" ) );
+        event.setCompletedBy( rs.getString( "completedby" ) );
+        event.setCompletedDate( DateUtils.getIso8601NoTz( rs.getDate( "completeddate" ) ) );
+        event.setCreated( DateUtils.getIso8601NoTz( rs.getDate( "created" ) ) );
+        event.setCreatedAtClient( DateUtils.getIso8601NoTz( rs.getDate( "createdatclient" ) ) );
+        event.setLastUpdated( DateUtils.getIso8601NoTz( rs.getDate( "lastupdated" ) ) );
+        event.setLastUpdatedAtClient( DateUtils.getIso8601NoTz( rs.getDate( "lastupdatedatclient" ) ) );
+
+        resolveGeometry( rs.getBytes( "geometry" ) ).ifPresent( event::setGeometry );
+
+        event.setDeleted( rs.getBoolean( "deleted" ) );
+
+        event.setProgram( rs.getString("prguid") );
+        //event.setEnrollment( programStageInstance.getProgramInstance().getUid() ); TODO do we need to duplicate this value? we have already setTrackedEntityInstance
+        event.setProgramStage( rs.getString("prgstguid") );
+        event.setAttributeOptionCombo( rs.getString("cocuid") );
+        event.setAttributeCategoryOptions( rs.getString("catoptions") );
+
+//        if ( programStageInstance.getProgramInstance().getEntityInstance() != null )
```

Comment:
Commented code.

Reply:
This is a TODO, will remove it asap, when the TODO is cleared